docs/explanation/gno-modules.md

```diff
@@ -0,0 +1,47 @@
+---
+id: gno-modules
+---
+
+# Gno Modules
+
+The packages and realms containing `gno.mod` file can be reffered as Gno modules. `gno.mod` file is introduced to enhance local testing and handle dependency management while testing Gno packages/realms locally. While it primarily serves local development and testing purposes, blockchain do not recognize or utilize this file directly.
+
+## What Is the gno.mod File for?
+
+`gno.mod` file is a critical for local testing and development. Its primary purposes include:
+
+- **Local Dependency Management**: The gno.mod file allows you to manage local dependencies effectively when developing God Modules. This facilitates testing and iterative development.
+
+- **Module Sorting while Publishing**: Additionally, the gno.mod file is used to sort and automatically publish modules that are located within the `/examples` directory to the blockchain when the chain starts.
+
+- **Configuration and Metadata (Potential Future Use)**: While the gno.mod file is currently used for specifying dependencies, it's worth noting that in the future, it might also serve as a container for additional configuration and metadata related to Gno Modules. This could include information such as module descriptions, veriosn, authorship details, or licensing information. (See: https://github.com/gnolang/gno/issues/498)
+
+## Gno Modules and Subdirectories
+
+It's important to note that Gno Modules do not include subdirectories. Each directory within your project is treated as an individual Gno Module, and each should contain its own gno.mod file, even if it's located within an existing Gno Module directory.
+
```

Comment:
```suggestion
```

[citation needed]

Ie. too much TBD to document this right now IMO.

Reply:
I think we should keep it as is in the PR for now, we've had good content on it, at least as a starting point.
We can always refine it in a future PR with @harry-hov. What do you think?